### PR TITLE
add 'Also known as', AKA 'AKA', to module references

### PR DESCRIPTION
**Pull request by busterb**
This adds a new reference type for things like named vulns, nicknames, marketing speak, etc. to make it possible to add a reference for common nomenclature without having to use these names in the modules themselves.

## Verification

- [x] Start `msfconsole`
- [x] `modules/auxiliary/server/dhclient_bash_env`
- [x] `info`
- [x] **Verify** that an 'Also known as:' annotation appears

### CODE REVIEW

File: modules/exploits/multi/misc/legend_bot_exec.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def initialize(info = {})
       'License'        => MSF_LICENSE,
       'References'     =>
         [
+          [ 'AKA', 'Shellshock' ],
```
Comment (wvu-r7) [inline modules/exploits/multi/misc/legend_bot_exec.rb]: This is an exploit for the bot, so there's no Shellshock here. The bot was likely deployed through other vulns, too.

File: modules/exploits/multi/misc/xdh_x_exec.rb [hunk 2]
```diff
@@ -33,6 +33,7 @@ def initialize(info = {})
       'License'        => MSF_LICENSE,
       'References'     =>
         [
+          [ 'AKA', 'Shellshock' ],
```
Comment (wvu-r7) [inline modules/exploits/multi/misc/xdh_x_exec.rb]: Same as above. Bot exploit.

File: modules/exploits/windows/iis/iis_webdav_scstoragepathfromurl.rb [hunk 3]
```diff
@@ -34,6 +34,7 @@ def initialize(info = {})
       'License'        => MSF_LICENSE,
       'References'     =>
         [
+          [ 'AKA', 'explodingcan' ],
```
Comment (wvu-r7) [inline modules/exploits/windows/iis/iis_webdav_scstoragepathfromurl.rb]: Can you capitalize this like the other NSA code names? Thanks!

**Comment by wvu-r7** [comment]
So, the nice thing about this that I wanted to test is that it works with `search` seamlessly and without being case-sensitive.

```
msf > search explodingcan

Matching Modules
================

   Name                                                 Disclosure Date  Rank    Description
   ----                                                 ---------------  ----    -----------
   exploit/windows/iis/iis_webdav_scstoragepathfromurl  2017-03-26       manual   Microsoft IIS WebDav ScStoragePathFromUrl Overflow


msf > 
```

Double win.

**Comment by h00die** [comment]
+1 this request, its a good idea to pull these things out.

**Comment by wvu-r7** [comment]
# Release Notes

This adds a new module reference type `AKA` (also known as) to map modules to a particular vulnerability's common name.

**Pull request merged**